cpp/include/rapidsmpf/streaming/core/message.hpp

```diff
@@ -26,64 +27,39 @@ namespace rapidsmpf::streaming {
     /**
-     * @brief Callback functions associated with a message.
+     * @brief Callback for performing a deep copy of a message.
      *
-     * Allows type-specific customization of memory-related operations, such as
-     * computing data sizes or performing deep copies.
+     * The copy operation allocates new memory for the message's payload using the
+     * provided buffer resource and memory reservation. The memory type specified
+     * in the reservation determines where the new copy will primarily reside
+     * (e.g., device or host memory).
+     *
+     * @param msg Source message to copy.
+     * @param br Buffer resource used for memory allocations.
+     * @param reservation Memory reservation to consume during allocation.
+     * @return A new `Message` instance containing a deep copy of the payload.
      */
-    struct Callbacks {
-        /**
-         * @brief Callback for computing the size of a message's content.
-         *
-         * This callback returns the total size, in bytes, of the data portion associated
-         * with a message. It is used to determine memory requirements for the content
-         * only — not for metadata or any auxiliary information.
-         *
-         * Typically, this data is represented by a `Buffer` that may reside in any
-         * memory type (e.g., host or device). The size reported by this callback
-         * determines how large a memory reservation must be when performing a
-         * `Message::copy()`.
-         *
-         * @param msg Reference to the message whose data size is queried.
-         * @param mem_type Target memory type to query.
-         * @return A pair (size, spillable) where:
-         *   - size: total size (in bytes) of the content for the given memory type.
-         *   - spillable: `true` if the message owns its buffers and releasing it frees
-         *     memory; otherwise `false`.
-         */
-        std::function<std::pair<size_t, bool>(Message const&, MemoryType)> content_size;
-
-        /**
-         * @brief Callback for performing a deep copy of a message.
-         *
-         * The copy operation allocates new memory for the message's payload using the
-         * provided buffer resource and memory reservation. The memory type specified
-         * in the reservation determines where the new copy will primarily reside
-         * (e.g., device or host memory).
-         *
-         * @param msg Source message to copy.
-         * @param br Buffer resource used for memory allocations.
-         * @param reservation Memory reservation to consume during allocation.
-         * @return A new `Message` instance containing a deep copy of the payload.
-         */
-        std::function<
-            Message(Message const&, BufferResource* br, MemoryReservation& reservation)>
-            copy;
-    };
+    using CopyCallback = std::function<
+        Message(Message const&, BufferResource* br, MemoryReservation& reservation)>;
```

Comment:
I think the `reservation` contains the relevant buffer resource pointer. So this callback should _only_ take the reservation (not the resource). In case we mix up and pass the wrong resource:

```suggestion
    using CopyCallback = std::function<
        Message(Message const&, MemoryReservation& reservation)>;
```